ecommerce/android/Getflix/app/src/main/java/com/example/getflix/services/ProductsAPI.kt

```diff
@@ -1,16 +1,20 @@
 package com.example.getflix.services
 
 import com.example.getflix.models.PModel
-import com.example.getflix.models.ProductModel
 import retrofit2.Response
 import retrofit2.http.GET
+import retrofit2.http.Path
 
 interface ProductsAPI {
 
-     @GET("products/homepage/3")
-     suspend fun getProducts(): Response<List<PModel>>
+     @GET("products/homepage/{numberOfProducts}")
+     suspend fun getProducts(@Path("numberOfProducts") numberOfProducts: Int): Response<List<PModel>>
 
+     @GET("product/{productId}")
+     suspend fun getProduct(@Path("productId") id: Int): Response<List<PModel>>
```

Comment:
To be honest, I don't understand why you created a distinct class which uses the same api but different endpoints. Therefore, I think we should see eye to eye on which service,either GetflixApiService.kt or ProductsAPI.kt, we will be using and then unite all functions under this class instead one to one match between an endpoint and a class.

Reply:
moved to GetflixApiService now